components/board/customlinux/board.go

```diff
@@ -0,0 +1,143 @@
+//go:build linux
+
+// Package customlinux implements a board running linux
+// This is an Experimental package
+package customlinux
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"periph.io/x/host/v3"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/board/genericlinux"
+	"go.viam.com/rdk/resource"
+)
+
+func init() {
+	if _, err := host.Init(); err != nil {
+		golog.Global().Debugw("error initializing host", "error", err)
+	}
+
+	RegisterCustomBoard("customlinux")
+}
+
+// customLinuxBoard wraps the genericlinux board type so that both can implement their own Reconfigure function.
+type customLinuxBoard struct {
+	*genericlinux.SysfsBoard
+	mu     sync.RWMutex
+	logger golog.Logger
+}
+
+// RegisterCustomBoard registers a sysfs based board using the pin mappings.
+func RegisterCustomBoard(modelName string) {
+	resource.RegisterComponent(
+		board.API,
+		resource.DefaultModelFamily.WithModel(modelName),
+		resource.Registration[board.Board, *Config]{
+			Constructor: createNewBoard,
+		})
+}
+
+func createNewBoard(
+	ctx context.Context,
+	_ resource.Dependencies,
+	conf resource.Config,
+	logger golog.Logger,
+) (board.Board, error) {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return nil, err
+	}
+
+	pinDefs, err := parseBoardConfig(newConf.PinConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	gpioMappings, err := genericlinux.GetGPIOBoardMappingFromPinDefs(pinDefs)
+	if err != nil {
+		return nil, err
+	}
+
+	boardConfig := genericlinux.Config{
+		I2Cs:              newConf.I2Cs,
+		SPIs:              newConf.SPIs,
+		Analogs:           newConf.Analogs,
+		DigitalInterrupts: newConf.DigitalInterrupts,
+	}
+	b, err := genericlinux.NewSysfsBoard(ctx, conf.ResourceName().AsNamed(), &boardConfig, gpioMappings, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	gb, ok := b.(*genericlinux.SysfsBoard)
+	if !ok {
+		return nil, errors.New("error creating board object")
+	}
+	return &customLinuxBoard{SysfsBoard: gb, logger: logger}, nil
```

Comment:
i guess it doesn't hurt anything but it also doesn't do anything- the Reconfigure of the genericlinux board is called in NewSysfsBoard and the customlinux Reconfigure just wraps that, so it would just be redundant. i can add a comment explaning that reconfigure is called when creating the genericlinux board